src/omega_red/redis/protocol.clj

```diff
@@ -48,17 +48,17 @@
   [^JedisPooled client cmds+args]
   {:pre [(seq cmds+args)
          (every? keyword? (map first cmds+args))]}
-  (let [pipeline ^AbstractPipeline (.pipelined client)
-        responses (mapv (fn [cmd+args]
-                          (let [[proto-command command-args] (cmd+args->command-with-args cmd+args)]
-                            (AbstractPipeline/.sendCommand pipeline
-                                                           ^Protocol$Command proto-command
-                                                           ^String/1 command-args)))
-                        cmds+args)]
-    (.sync pipeline)
-    (->> responses
-         (mapv Response/.get)
-         (mapv codec/deserialize))))
+  (with-open [pipeline ^AbstractPipeline (.pipelined client)]
```

Comment:
This was the bug I introduced - I assumed that because we are always using `JedisPooled`, acquire/release will happen automatically, turns out we need to explicitly close the pipeline. Not sure how this worked before in my previous project because I didn't run into any timeouts etc, but then again - there were many other factors (different cloud provider, size, request volume)